### ISSUE TITLE
Any POST request fails (invalid request) and error object is nil

### ISSUE THREAD

**Issue by javierjulio** [synapsepayments/synapse_pay-ruby#4]
Running the following code (taken from code samples doc) in a console with sandbox credentials fails with a 400 Invalid Request error with a message of "Please check the URL and parameters.":

``` ruby
client = SynapsePay::User.create({
  :email => "test-user@synapsepay.com",
  :fullname => "Test Account",
  :ip_address => "11.111.11.11",
  :phonenumber => "123456789",
  :force_create => "no"
})

begin
  bank = client.banks.add({
    :fullname => "Jon Smith",
    :account_num => "1111111111",
    :routing_num => "121000358",
    :nickname => "Example bank account",
    :account_type => "1",
    :account_class => "1"
  })
rescue SynapsePay::APIError => e
  puts e
end
```

**Comment by javierjulio** [comment]
Further the error object is nil. 😞 I installed httplog gem and can provide the logs if necessary. The JSON returned for adding a bank account (which failed) came back as the following:

``` json
{"reason": "Incorrect CONTENT_TYPE. Only application/json accepted", "success": false}
```

so it seems the gem isn't POSTing to the banks/add resource with JSON. The httplog shows all the params going through as query params instead.

**Comment by javierjulio** [comment]
Finding out now that any POST request through the gem fails. The gem seems to set the right content-type header (HttpLog shows two though, the latter one has a value of application/json) but its creating a query string rather than a data set. I don't believe the Requester object is correctly picking up that the POST request contains JSON so its constructing a query param string instead.

**Comment by javierjulio** [comment]
Ok I've pinpointed the reason for this is that somewhere in the code base the headers hash isn't consistent with strings and symbols. In the `Requester.prepare_params` method when it checks for the Content-Type header its not always a string as in some cases (in my case for adding a bank account) its a symbol.

**Comment by javierjulio** [comment]
This [line of code in APIClient](https://github.com/synapsepayments/synapse_pay-ruby/blob/master/lib/synapse_pay/apibits/api_client.rb#L16) is the culprit as its reusing ParamsBuilder to merge headers but part of that process is also symbolizing any keys thus it changes `'Content-Type'` to `:'Content-Type'` and then the check in `Requester.prepare_params` fails.

Do you want to symbolize headers or leave them as is? If so I suggest doing that in the HeadersBuilder or separating out the merge/clean process from ParamsBuilder into a Util and use in both. Happy to provide a PR if you know what you prefer.

**Comment by joncalhoun** [comment]
Thanks for digging into this. I'll spend some time fixing this and get it all worked out. We used to use purely symbols but had opted to use strings (eg "Content-Type") in the headers in that case so it matched between libs but it looks like we missed a case where this is converted to a symbol. :frowning:

**Comment by dkarter** [comment]
@joncalhoun Can you release a new official version? I'd point my gemfile to master but I'm afraid it may get unstable.

**Comment by javierjulio** [comment]
@dkarter this issue has already been resolved with the latest stable release, 0.0.3 on rubygems.
